diff --git a/system.prop b/system.prop
index 4defb39..14fb8b2 100644
--- a/system.prop
+++ b/system.prop
@@ -50,7 +50,7 @@ persist.vendor.overlay.izat.optin=rro
 debug.hwui.skia_atrace_enabled=false
 dev.pm.dyn_samplingrate=1
 persist.demo.hdmirotationlock=false
-sdm.debug.disable_skip_validate=0
+sdm.debug.disable_skip_validate=1
 
 # IMS
 persist.dbg.ims_volte_enable=1
@@ -170,6 +170,8 @@ persist.bt.a2dp.aac_disable=true
 # System property for cabl
 ro.qualcomm.cabl=2
 ro.vendor.display.cabl=2
+config.cabl.xml=1
+config.cabl.path=/vendor/etc/CABLConfig.xml
 
 #
 # System props for telephony
