diff --git a/proprietary/vendor/etc/thermal-engine.conf b/proprietary/vendor/etc/thermal-engine.conf
index cee84e2..9b9be5e 100644
--- a/proprietary/vendor/etc/thermal-engine.conf
+++ b/proprietary/vendor/etc/thermal-engine.conf
@@ -21,16 +21,16 @@ algo_type ss
 sampling 1000
 sensor tsens_tz_sensor12
 device gpu
-set_point 70000
-set_point_clr 60000
+set_point 80000
+set_point_clr 70000
 time_constant 0
 
 [SKIN-MONITOR]
 algo_type monitor
 sampling 5000
 sensor tsens_tz_sensor5
-thresholds 50000 55000 60000
-thresholds_clr 47000 52000 57000
+thresholds 60000 65000 70000
+thresholds_clr 57000 62000 67000
 actions cpu0+cpu4+gpu cpu0+cpu4+gpu cpu0+cpu4+gpu
 action_info 1401600+1401600+510000000 1113600+1113600+400000000 902400+902400+320000000
 
