diff --git a/configs/keylayout/idc/stmpe_azerty_keypad.idc b/configs/keylayout/idc/stmpe_azerty_keypad.idc
index 9197f76..775c360 100644
--- a/configs/keylayout/idc/stmpe_azerty_keypad.idc
+++ b/configs/keylayout/idc/stmpe_azerty_keypad.idc
@@ -1,32 +1,13 @@
-# Copyright (C) 2015 The Android Open Source Project
-#
-# Licensed under the Apache License, Version 2.0 (the "License");
-# you may not use this file except in compliance with the License.
-# You may obtain a copy of the License at
-#
-#      http://www.apache.org/licenses/LICENSE-2.0
-#
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
+# Copyright (c) 2015 by BlackBerry Limited. All Rights Reserved.
+# BlackBerry Limited does not grant any rights,
+# including the accompanying associated documentation,
+# (the Software) for any purpose whatsoever, including without limitation
+# any rights to copy, use or distribute the Software. Your rights to use the
+# Software shall be only as set forth in any separate license between you and
+# BlackBerry Limited.
 
-#
-# BlackBerry Key2 input device configuration file for the built-in keyboard
-#
-
-# The keyboard is internal because it is part of the device.
 device.internal = 1
 
-# The keyboard is the default built-in keyboard so it should be assigned
-# an id of 0.
 keyboard.builtIn = 1
-
-# The keyboard is always used in landscape orientation, so don't rotate
-# the arrow keys to compensate for orientation changes
-keyboard.orientationAware = 0
-
-# Our custom key character map
+keyboard.layout = stmpe_azerty
 keyboard.characterMap = stmpe
-keyboard.layout = stmpe
diff --git a/configs/keylayout/idc/stmpe_keypad.idc b/configs/keylayout/idc/stmpe_keypad.idc
index 9197f76..a8793c5 100644
--- a/configs/keylayout/idc/stmpe_keypad.idc
+++ b/configs/keylayout/idc/stmpe_keypad.idc
@@ -1,32 +1,13 @@
-# Copyright (C) 2015 The Android Open Source Project
-#
-# Licensed under the Apache License, Version 2.0 (the "License");
-# you may not use this file except in compliance with the License.
-# You may obtain a copy of the License at
-#
-#      http://www.apache.org/licenses/LICENSE-2.0
-#
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
+# Copyright (c) 2015 by BlackBerry Limited. All Rights Reserved.
+# BlackBerry Limited does not grant any rights,
+# including the accompanying associated documentation,
+# (the Software) for any purpose whatsoever, including without limitation
+# any rights to copy, use or distribute the Software. Your rights to use the
+# Software shall be only as set forth in any separate license between you and
+# BlackBerry Limited.
 
-#
-# BlackBerry Key2 input device configuration file for the built-in keyboard
-#
-
-# The keyboard is internal because it is part of the device.
 device.internal = 1
 
-# The keyboard is the default built-in keyboard so it should be assigned
-# an id of 0.
 keyboard.builtIn = 1
-
-# The keyboard is always used in landscape orientation, so don't rotate
-# the arrow keys to compensate for orientation changes
-keyboard.orientationAware = 0
-
-# Our custom key character map
-keyboard.characterMap = stmpe
 keyboard.layout = stmpe
+keyboard.characterMap = stmpe
diff --git a/configs/keylayout/idc/stmpe_qwertz_keypad.idc b/configs/keylayout/idc/stmpe_qwertz_keypad.idc
index 9197f76..e229dd9 100644
--- a/configs/keylayout/idc/stmpe_qwertz_keypad.idc
+++ b/configs/keylayout/idc/stmpe_qwertz_keypad.idc
@@ -1,32 +1,13 @@
-# Copyright (C) 2015 The Android Open Source Project
-#
-# Licensed under the Apache License, Version 2.0 (the "License");
-# you may not use this file except in compliance with the License.
-# You may obtain a copy of the License at
-#
-#      http://www.apache.org/licenses/LICENSE-2.0
-#
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
+# Copyright (c) 2015 by BlackBerry Limited. All Rights Reserved.
+# BlackBerry Limited does not grant any rights,
+# including the accompanying associated documentation,
+# (the Software) for any purpose whatsoever, including without limitation
+# any rights to copy, use or distribute the Software. Your rights to use the
+# Software shall be only as set forth in any separate license between you and
+# BlackBerry Limited.
 
-#
-# BlackBerry Key2 input device configuration file for the built-in keyboard
-#
-
-# The keyboard is internal because it is part of the device.
 device.internal = 1
 
-# The keyboard is the default built-in keyboard so it should be assigned
-# an id of 0.
 keyboard.builtIn = 1
-
-# The keyboard is always used in landscape orientation, so don't rotate
-# the arrow keys to compensate for orientation changes
-keyboard.orientationAware = 0
-
-# Our custom key character map
+keyboard.layout = stmpe_qwertz
 keyboard.characterMap = stmpe
-keyboard.layout = stmpe
diff --git a/configs/keylayout/keychars/stmpe.kcm b/configs/keylayout/keychars/stmpe.kcm
index 3bee186..e5fc33d 100644
--- a/configs/keylayout/keychars/stmpe.kcm
+++ b/configs/keylayout/keychars/stmpe.kcm
@@ -1,360 +1,247 @@
-# Copyright (C) 2010 The Android Open Source Project
-#
-# Licensed under the Apache License, Version 2.0 (the "License");
-# you may not use this file except in compliance with the License.
-# You may obtain a copy of the License at
-#
-#      http://www.apache.org/licenses/LICENSE-2.0
-#
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
+# Copyright (C) 2015 BlackBerry Limited
 
-type ALPHA
-
-### Basic QWERTY keys ###
+# Blackberry Physical Keyboard
 
-key A {
-    label:                              'A'
-    base:                               'a'
-    shift, capslock:                    'A'
-    shift+capslock:                     'a'
-}
+type ALPHA
 
-key B {
-    label:                              'B'
-    base:                               'b'
-    shift, capslock:                    'B'
-    shift+capslock:                     'b'
-}
+### ROW 1
 
-key C {
-    label:                              'C'
-    base:                               'c'
-    shift, capslock:                    'C'
-    alt:                                '\u00e7'
-    shift+alt, alt+capslock:            '\u00c7'
-    shift+capslock:                     'c'
+key Q {
+    label:                              'Q'
+    base:                               'q'
+    shift, capslock:                    'Q'
+    alt, lalt:                          '#'
+    sym:                                '~'
 }
 
-key D {
-    label:                              'D'
-    base:                               'd'
-    shift, capslock:                    'D'
-    shift+capslock:                     'd'
+key W {
+    label:                              'W'
+    base:                               'w'
+    shift, capslock:                    'W'
+    alt, lalt:                          '1'
+    sym:                                '`'
 }
 
 key E {
     label:                              'E'
     base:                               'e'
     shift, capslock:                    'E'
-    alt:                                '\u0301'
-    shift+capslock:                     'e'
+    alt, lalt:                          '2'
+    sym:                                '{'
 }
 
-key F {
-    label:                              'F'
-    base:                               'f'
-    shift, capslock:                    'F'
-    shift+capslock:                     'f'
+key R {
+    label:                              'R'
+    base:                               'r'
+    shift, capslock:                    'R'
+    alt, lalt:                          '3'
+    sym:                                '}'
 }
 
-key G {
-    label:                              'G'
-    base:                               'g'
-    shift, capslock:                    'G'
-    shift+capslock:                     'g'
+key T {
+    label:                              'T'
+    base:                               't'
+    shift, capslock:                    'T'
+    alt, lalt:                          '('
+    sym:                                '['
 }
 
-key H {
-    label:                              'H'
-    base:                               'h'
-    shift, capslock:                    'H'
-    shift+capslock:                     'h'
+key Y {
+    label:                              'Y'
+    base:                               'y'
+    shift, capslock:                    'Y'
+    alt, lalt:                          ')'
+    sym:                                ']'
+}
+
+key U {
+    label:                              'U'
+    base:                               'u'
+    shift, capslock:                    'U'
+    alt, lalt:                          '_'
+    sym:                                '<'
 }
 
 key I {
     label:                              'I'
     base:                               'i'
     shift, capslock:                    'I'
-    alt:                                '\u0302'
-    shift+capslock:                     'i'
-}
-
-key J {
-    label:                              'J'
-    base:                               'j'
-    shift, capslock:                    'J'
-    shift+capslock:                     'j'
-}
-
-key K {
-    label:                              'K'
-    base:                               'k'
-    shift, capslock:                    'K'
-    shift+capslock:                     'k'
-}
-
-key L {
-    label:                              'L'
-    base:                               'l'
-    shift, capslock:                    'L'
-    shift+capslock:                     'l'
-    fn:                                 ';'
-}
-
-key M {
-    label:                              'M'
-    base:                               'm'
-    shift, capslock:                    'M'
-    shift+capslock:                     'm'
-}
-
-key N {
-    label:                              'N'
-    base:                               'n'
-    shift, capslock:                    'N'
-    alt:                                '\u0303'
-    shift+capslock:                     'n'
+    alt, lalt:                          '-'
+    sym:                                '>'
 }
 
 key O {
     label:                              'O'
     base:                               'o'
     shift, capslock:                    'O'
-    shift+capslock:                     'o'
+    alt, lalt:                          '+'
+    sym:                                '^'
 }
 
 key P {
     label:                              'P'
     base:                               'p'
     shift, capslock:                    'P'
-    shift+capslock:                     'p'
-    fn:                                 '\\'
+    alt, lalt:                          '@'
+    sym:                                '%'
 }
 
-key Q {
-    label:                              'Q'
-    base:                               'q'
-    shift, capslock:                    'Q'
-    shift+capslock:                     'q'
-}
+### ROW 2
 
-key R {
-    label:                              'R'
-    base:                               'r'
-    shift, capslock:                    'R'
-    shift+capslock:                     'r'
+key A {
+    label:                              'A'
+    base:                               'a'
+    shift, capslock:                    'A'
+    alt, lalt:                          '*'
+    sym:                                '='
 }
 
 key S {
     label:                              'S'
     base:                               's'
     shift, capslock:                    'S'
-    alt:                                '\u00df'
-    shift+capslock:                     's'
+    alt, lalt:                          '4'
+    sym:                                '\u00F7'
 }
 
-key T {
-    label:                              'T'
-    base:                               't'
-    shift, capslock:                    'T'
-    shift+capslock:                     't'
+key D {
+    label:                              'D'
+    base:                               'd'
+    shift, capslock:                    'D'
+    alt, lalt:                          '5'
+    sym:                                '\u00B1'
 }
 
-key U {
-    label:                              'U'
-    base:                               'u'
-    shift, capslock:                    'U'
-    alt:                                '\u0308'
-    shift+capslock:                     'u'
+key F {
+    label:                              'F'
+    base:                               'f'
+    shift, capslock:                    'F'
+    alt, lalt:                          '6'
+    sym:                                '\u2022'
 }
 
-key V {
-    label:                              'V'
-    base:                               'v'
-    shift, capslock:                    'V'
-    shift+capslock:                     'v'
+key G {
+    label:                              'G'
+    base:                               'g'
+    shift, capslock:                    'G'
+    alt, lalt:                          '/'
+    sym:                                '\\'
 }
 
-key W {
-    label:                              'W'
-    base:                               'w'
-    shift, capslock:                    'W'
-    shift+capslock:                     'w'
+key H {
+    label:                              'H'
+    base:                               'h'
+    shift, capslock:                    'H'
+    alt, lalt:                          ':'
+    sym:                                '|'
 }
 
-key X {
-    label:                              'X'
-    base:                               'x'
-    shift, capslock:                    'X'
-    shift+capslock:                     'x'
+key J {
+    label:                              'J'
+    base:                               'j'
+    shift, capslock:                    'J'
+    alt, lalt:                          ';'
+    sym:                                '&'
 }
 
-key Y {
-    label:                              'Y'
-    base:                               'y'
-    shift, capslock:                    'Y'
-    shift+capslock:                     'y'
+key K {
+    label:                              'K'
+    base:                               'k'
+    shift, capslock:                    'K'
+    alt, lalt:                          '\''
+    sym:                                '\u201C'
+}
+
+key L {
+    label:                              'L'
+    base:                               'l'
+    shift, capslock:                    'L'
+    alt, lalt:                          '\"'
+    sym:                                '\u201D'
 }
 
+### ROW 3
+
 key Z {
     label:                              'Z'
     base:                               'z'
     shift, capslock:                    'Z'
-    shift+capslock:                     'z'
-}
-
-key 0 {
-    label:                              '0'
-    base:                               '0'
-    shift, fn:                          ')'
+    alt, lalt:                          '7'
+    sym:                                '\u00A5'
 }
 
-key 1 {
-    label:                              '1'
-    base:                               '1'
-    shift, fn:                          '!'
+key X {
+    label:                              'X'
+    base:                               'x'
+    shift, capslock:                    'X'
+    alt, lalt:                          '8'
+    sym:                                '\u20AC'
 }
 
-key 2 {
-    label:                              '2'
-    base:                               '2'
-    shift, fn:                          '@'
+key C {
+    label:                              'C'
+    base:                               'c'
+    shift, capslock:                    'C'
+    alt, lalt:                          '9'
+    sym:                                '\u00A3'
 }
 
-key 3 {
-    label:                              '3'
-    base:                               '3'
-    shift, fn:                          '#'
+key V {
+    label:                              'V'
+    base:                               'v'
+    shift, capslock:                    'V'
+    alt, lalt:                          '?'
+    sym:                                '\u00BF'
 }
 
-key 4 {
-    label:                              '4'
-    base:                               '4'
-    shift, fn:                          '$'
+key B {
+    label:                              'B'
+    base:                               'b'
+    shift, capslock:                    'B'
+    alt, lalt:                          '!'
+    sym:                                '\u00A1'
 }
 
-key 5 {
-    label:                              '5'
-    base:                               '5'
-    shift, fn:                          '%'
+key N {
+    label:                              'N'
+    base:                               'n'
+    shift, capslock:                    'N'
+    alt, lalt:                          ','
+    sym:                                '\u00AB'
 }
 
-key 6 {
-    label:                              '6'
-    base:                               '6'
-    shift, fn:                          '^'
-    shift+alt, alt+capslock:            '\u0302'
+key M {
+    label:                              'M'
+    base:                               'm'
+    shift, capslock:                    'M'
+    alt, lalt:                          '.'
+    sym:                                '\u00BB'
 }
 
-key 7 {
-    label:                              '7'
-    base:                               '7'
-    shift, fn:                          '&'
+key 4 {
+    label:                              '$'
+    base, capslock:                     '$'
+    shift:                              '\u20AC' # Euro
+    alt, lalt:                          '$'
+    sym:                                '$'
+    lalt+lshift:                        '\uE141' # Special character means this is speaker key
 }
 
-key 8 {
-    label:                              '8'
-    base:                               '8'
-    shift, fn:                          '*'
-}
+### ROW 4
 
-key 9 {
-    label:                              '9'
-    base:                               '9'
-    shift, fn:                          '('
+key 0 {
+    label:                              '0'
+    base:                               '0'
+    shift, capslock:                    '0'
+    alt, lalt:                          '0'
 }
 
 key SPACE {
     label:                              ' '
-    base:                               ' '
-    alt, meta:                          fallback SEARCH
-    ctrl, fn:                           fallback LANGUAGE_SWITCH
+    base, shift, capslock, alt, lalt:   ' '
 }
 
 key ENTER {
     label:                              '\n'
     base:                               '\n'
 }
-
-# The alt-key layout maps special characters to F1-F10 keys
-key F1 {
-    label:                              '#'
-    base:                               '#'
-}
-
-key F2 {
-    label:                              '('
-    base:                               '('
-}
-
-key F3 {
-    label:                              ')'
-    base:                               ')'
-}
-
-key F4 {
-    label:                              '_'
-    base:                               '_'
-}
-
-# Minus is handled correctly by Android Framework
-
-key NUMPAD_ADD {
-    label:                              '+'
-    base:                               '+'
-}
-
-key F5 {
-    label:                              '@'
-    base:                               '@'
-}
-
-key NUMPAD_MULTIPLY {
-    label:                              '*'
-    base:                               '*'
-}
-
-key F6 {
-    label:                              ':'
-    base:                               ':'
-}
-
-key F7 {
-    label:                              '"'
-    base:                               '"'
-}
-
-key PLUS {
-    label:                              '?'
-    base:                               '?'
-
-}
-
-key F8 {
-    label:                              '!'
-    base:                               '!'
-}
-
-# F10 is the "Assistant key" or something
-key F10 {
-    label:                              '"'
-    base:                               '"'
-}
-
-# F11 is SYM
-
-# F12 is Currency Key
-key F12 {
-    label:                              '\u0024'
-    base:                               '\u0024'
-}
-
-# F9 is ALT+CK
-key F9 {
-    label:                              '\u20ac'
-    base:                               '\u20ac'
-}
diff --git a/configs/keylayout/keylayout/stmpe.kl b/configs/keylayout/keylayout/stmpe.kl
index 7b01335..02c9f8f 100644
--- a/configs/keylayout/keylayout/stmpe.kl
+++ b/configs/keylayout/keylayout/stmpe.kl
@@ -1,499 +1,42 @@
-# Copyright (C) 2010 The Android Open Source Project
+# Copyright (C) 2015 BlackBerry Limited
 #
-# Licensed under the Apache License, Version 2.0 (the "License");
-# you may not use this file except in compliance with the License.
-# You may obtain a copy of the License at
+# Blackberry Physical Keyboard
 #
-#      http://www.apache.org/licenses/LICENSE-2.0
-#
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
 
-#
-# Generic key layout file for full alphabetic US English PC style external keyboards.
-#
-# This file is intentionally very generic and is intended to support a broad range of keyboards.
-# Do not edit the generic key layout to support a specific keyboard; instead, create
-# a new key layout file with the required keyboard configuration.
-#
-
-key 1     ESCAPE
-key 2     1
-key 3     2
-key 4     3
 key 5     4
-key 6     5
-key 7     6
-key 8     7
-key 9     8
-key 10    9
 key 11    0
-key 12    MINUS
-key 13    EQUALS
 key 14    DEL
-key 15    TAB
 key 16    Q
 key 17    W
 key 18    E
 key 19    R
 key 20    T
 key 21    Y
 key 22    U
 key 23    I
 key 24    O
 key 25    P
-key 26    LEFT_BRACKET
-key 27    RIGHT_BRACKET
 key 28    ENTER
-key 29    CTRL_LEFT
 key 30    A
 key 31    S
 key 32    D
 key 33    F
 key 34    G
 key 35    H
 key 36    J
 key 37    K
 key 38    L
-key 39    SEMICOLON
-key 40    APOSTROPHE
-key 41    GRAVE
 key 42    SHIFT_LEFT
-key 43    BACKSLASH
 key 44    Z
 key 45    X
 key 46    C
 key 47    V
 key 48    B
 key 49    N
 key 50    M
-key 51    COMMA
-key 52    PERIOD
-key 53    SLASH
 key 54    SHIFT_RIGHT
-key 55    NUMPAD_MULTIPLY
 key 56    ALT_LEFT
 key 57    SPACE
-key 58    CAPS_LOCK
-key 59    F1
-key 60    F2
-key 61    F3
-key 62    F4
-key 63    F5
-key 64    F6
-key 65    F7
-key 66    F8
-key 67    F9
-key 68    F10
-key 69    NUM_LOCK
-key 70    SCROLL_LOCK
-key 71    NUMPAD_7
-key 72    NUMPAD_8
-key 73    NUMPAD_9
-key 74    NUMPAD_SUBTRACT
-key 75    NUMPAD_4
-key 76    NUMPAD_5
-key 77    NUMPAD_6
-key 78    NUMPAD_ADD
-key 79    NUMPAD_1
-key 80    NUMPAD_2
-key 81    NUMPAD_3
-key 82    NUMPAD_0
-key 83    NUMPAD_DOT
-# key 84 (undefined)
-key 85    ZENKAKU_HANKAKU
-key 86    BACKSLASH
-# key 87    (For SYM)
-key 88    F12
-key 89    RO
-# key 90 "KEY_KATAKANA"
-# key 91 "KEY_HIRAGANA"
-key 92    HENKAN
-key 93    KATAKANA_HIRAGANA
-key 94    MUHENKAN
-key 95    NUMPAD_COMMA
-key 96    NUMPAD_ENTER
-key 97    CTRL_RIGHT
-key 98    NUMPAD_DIVIDE
-key 99    SYSRQ
-key 100   ALT_RIGHT
-# key 101 "KEY_LINEFEED"
-key 102   MOVE_HOME
-key 103   DPAD_UP
-key 104   PAGE_UP
-key 105   DPAD_LEFT
-key 106   DPAD_RIGHT
-key 107   MOVE_END
-key 108   DPAD_DOWN
-key 109   PAGE_DOWN
-key 110   INSERT
-key 111   FORWARD_DEL
-# key 112 "KEY_MACRO"
-key 113   VOLUME_MUTE
-key 114   VOLUME_DOWN
-key 115   VOLUME_UP
-key 116   POWER
-key 117   NUMPAD_EQUALS
-# key 118 "KEY_KPPLUSMINUS"
-key 119   BREAK
-key 120   RECENT_APPS
-key 121   NUMPAD_COMMA
-key 122   KANA
-key 123   EISU
-key 124   YEN
-key 125   META_LEFT
-key 126   META_RIGHT
-key 127   MENU
-key 128   MEDIA_STOP
-# key 129 "KEY_AGAIN"
-# key 130 "KEY_PROPS"
-# key 131 "KEY_UNDO"
-# key 132 "KEY_FRONT"
-key 133   COPY
-# key 134 "KEY_OPEN"
-key 135   PASTE
-# key 136 "KEY_FIND"
-key 137   CUT
-# key 138 "KEY_HELP"
-key 139   MENU
-key 140   CALCULATOR
-# key 141 "KEY_SETUP"
-key 142   SLEEP
+key 100   SYM
+key 110   FUNCTION
 key 143   WAKEUP
-# key 144 "KEY_FILE"
-# key 145 "KEY_SENDFILE"
-# key 146 "KEY_DELETEFILE"
-# key 147 "KEY_XFER"
-# key 148 "KEY_PROG1"
-# key 149 "KEY_PROG2"
-key 150   EXPLORER
-# key 151 "KEY_MSDOS"
-key 152   LOCK
-# key 153 "KEY_DIRECTION"
-# key 154 "KEY_CYCLEWINDOWS"
-key 155   ENVELOPE
-key 156   BOOKMARK
-# key 157 "KEY_COMPUTER"
-key 158   BACK
-key 159   FORWARD
-key 160   MEDIA_CLOSE
-key 161   MEDIA_EJECT
-key 162   MEDIA_EJECT
-key 163   MEDIA_NEXT
-key 164   MEDIA_PLAY_PAUSE
-key 165   MEDIA_PREVIOUS
-key 166   MEDIA_STOP
-key 167   MEDIA_RECORD
-key 168   MEDIA_REWIND
-key 169   CALL
-# key 170 "KEY_ISO"
-key 171   MUSIC
-key 172   HOME
-key 173   REFRESH
-# key 174 "KEY_EXIT"
-# key 175 "KEY_MOVE"
-# key 176 "KEY_EDIT"
-key 177   PAGE_UP
-key 178   PAGE_DOWN
-key 179   NUMPAD_LEFT_PAREN
-key 180   NUMPAD_RIGHT_PAREN
-key 181   NEW
-# key 182 "KEY_REDO"
-key 183   F13
-key 184   F14
-key 185   F15
-key 186   F16
-key 187   F17
-key 188   F18
-key 189   F19
-key 190   F20
-key 191   F21
-key 192   F22
-key 193   F23
-key 194   F24
-# key 195 (undefined)
-# key 196 (undefined)
-# key 197 (undefined)
-# key 198 (undefined)
-# key 199 (undefined)
-key 200   MEDIA_PLAY
-key 201   MEDIA_PAUSE
-# key 202 "KEY_PROG3"
-# key 203 "KEY_PROG4"
-key 204 NOTIFICATION
-# key 205 "KEY_SUSPEND"
-key 206   CLOSE
-key 207   MEDIA_PLAY
-key 208   MEDIA_FAST_FORWARD
-# key 209 "KEY_BASSBOOST"
-key 210   PRINT
-# key 211 "KEY_HP"
-key 212   CAMERA
-key 213   MUSIC
-key 214   PLUS
-key 215   ENVELOPE
-# key 216 "KEY_CHAT"
-key 217   SEARCH
-# key 218 "KEY_CONNECT"
-# key 219 "KEY_FINANCE"
-# key 220 "KEY_SPORT"
-# key 221 "KEY_SHOP"
-# key 222 "KEY_ALTERASE"
-# key 223 "KEY_CANCEL"
-key 224   BRIGHTNESS_DOWN
-key 225   BRIGHTNESS_UP
-key 226   HEADSETHOOK
-key 228   KEYBOARD_BACKLIGHT_TOGGLE
-key 229   KEYBOARD_BACKLIGHT_DOWN
-key 230   KEYBOARD_BACKLIGHT_UP
-key 248   MUTE
-
-key 256   BUTTON_1
-key 257   BUTTON_2
-key 258   BUTTON_3
-key 259   BUTTON_4
-key 260   BUTTON_5
-key 261   BUTTON_6
-key 262   BUTTON_7
-key 263   BUTTON_8
-key 264   BUTTON_9
-key 265   BUTTON_10
-key 266   BUTTON_11
-key 267   BUTTON_12
-key 268   BUTTON_13
-key 269   BUTTON_14
-key 270   BUTTON_15
-key 271   BUTTON_16
-
-key 288   BUTTON_1
-key 289   BUTTON_2
-key 290   BUTTON_3
-key 291   BUTTON_4
-key 292   BUTTON_5
-key 293   BUTTON_6
-key 294   BUTTON_7
-key 295   BUTTON_8
-key 296   BUTTON_9
-key 297   BUTTON_10
-key 298   BUTTON_11
-key 299   BUTTON_12
-key 300   BUTTON_13
-key 301   BUTTON_14
-key 302   BUTTON_15
-key 303   BUTTON_16
-
-
-key 304   BUTTON_A
-key 305   BUTTON_B
-key 306   BUTTON_C
-key 307   BUTTON_X
-key 308   BUTTON_Y
-key 309   BUTTON_Z
-key 310   BUTTON_L1
-key 311   BUTTON_R1
-key 312   BUTTON_L2
-key 313   BUTTON_R2
-key 314   BUTTON_SELECT
-key 315   BUTTON_START
-key 316   BUTTON_MODE
-key 317   BUTTON_THUMBL
-key 318   BUTTON_THUMBR
-
-
-key 329   STYLUS_BUTTON_TERTIARY
-key 331   STYLUS_BUTTON_PRIMARY
-key 332   STYLUS_BUTTON_SECONDARY
-
-
-# key 352 "KEY_OK"
-key 353   DPAD_CENTER
-# key 354 "KEY_GOTO"
-# key 355 "KEY_CLEAR"
-# key 356 "KEY_POWER2"
-# key 357 "KEY_OPTION"
-# key 358 "KEY_INFO"
-# key 359 "KEY_TIME"
-# key 360 "KEY_VENDOR"
-# key 361 "KEY_ARCHIVE"
-key 362   GUIDE
-# key 363 "KEY_CHANNEL"
-# key 364 "KEY_FAVORITES"
-# key 365 "KEY_EPG"
-key 366   DVR
-# key 367 "KEY_MHP"
-key 368   LANGUAGE_SWITCH
-# key 369 "KEY_TITLE"
-key 370   CAPTIONS
-# key 371 "KEY_ANGLE"
-key 372   FULLSCREEN
-# key 373 "KEY_MODE"
-# key 374 "KEY_KEYBOARD"
-# key 375 "KEY_SCREEN"
-# key 376 "KEY_PC"
-key 377   TV
-# key 378 "KEY_TV2"
-# key 379 "KEY_VCR"
-# key 380 "KEY_VCR2"
-# key 381 "KEY_SAT"
-# key 382 "KEY_SAT2"
-# key 383 "KEY_CD"
-# key 384 "KEY_TAPE"
-# key 385 "KEY_RADIO"
-# key 386 "KEY_TUNER"
-# key 387 "KEY_PLAYER"
-# key 388 "KEY_TEXT"
-# key 389 "KEY_DVD"
-# key 390 "KEY_AUX"
-# key 391 "KEY_MP3"
-# key 392 "KEY_AUDIO"
-# key 393 "KEY_VIDEO"
-# key 394 "KEY_DIRECTORY"
-# key 395 "KEY_LIST"
-# key 396 "KEY_MEMO"
-key 397   CALENDAR
-key 398   PROG_RED
-key 399   PROG_GREEN
-key 400   PROG_YELLOW
-key 401   PROG_BLUE
-key 402   CHANNEL_UP
-key 403   CHANNEL_DOWN
-# key 404 "KEY_FIRST"
-key 405   LAST_CHANNEL
-# key 406 "KEY_AB"
-# key 407 "KEY_NEXT"
-# key 408 "KEY_RESTART"
-# key 409 "KEY_SLOW"
-# key 410 "KEY_SHUFFLE"
-# key 411 "KEY_BREAK"
-# key 412 "KEY_PREVIOUS"
-# key 413 "KEY_DIGITS"
-# key 414 "KEY_TEEN"
-# key 415 "KEY_TWEN"
-# key 418 "KEY_ZOOM_IN"
-key 418   ZOOM_IN
-# key 419 "KEY_ZOOM_OUT"
-key 419   ZOOM_OUT
-key 528 FOCUS
-
-key 429   CONTACTS
-
-# key 448 "KEY_DEL_EOL"
-# key 449 "KEY_DEL_EOS"
-# key 450 "KEY_INS_LINE"
-# key 451 "KEY_DEL_LINE"
-
-
-key 464   FUNCTION
-key 465   ESCAPE            FUNCTION
-key 466   F1                FUNCTION
-key 467   F2                FUNCTION
-key 468   F3                FUNCTION
-key 469   F4                FUNCTION
-key 470   F5                FUNCTION
-key 471   F6                FUNCTION
-key 472   F7                FUNCTION
-key 473   F8                FUNCTION
-key 474   F9                FUNCTION
-key 475   F10               FUNCTION
-key 476   F11               FUNCTION
-key 477   F12               FUNCTION
-key 478   1                 FUNCTION
-key 479   2                 FUNCTION
-key 480   D                 FUNCTION
-key 481   E                 FUNCTION
-key 482   F                 FUNCTION
-key 483   S                 FUNCTION
-key 484   B                 FUNCTION
-
-
-# key 497 KEY_BRL_DOT1
-# key 498 KEY_BRL_DOT2
-# key 499 KEY_BRL_DOT3
-# key 500 KEY_BRL_DOT4
-# key 501 KEY_BRL_DOT5
-# key 502 KEY_BRL_DOT6
-# key 503 KEY_BRL_DOT7
-# key 504 KEY_BRL_DOT8
-
-key 522   STAR
-key 523   POUND
-key 580   APP_SWITCH
-key 582   VOICE_ASSIST
-# Linux KEY_ASSISTANT
-key 583   ASSIST
-key 585   EMOJI_PICKER
-key 586   DICTATE
-key 656   MACRO_1
-key 657   MACRO_2
-key 658   MACRO_3
-key 659   MACRO_4
-
-# Keys defined by HID usages
-key usage 0x010082 LOCK                      FALLBACK_USAGE_MAPPING
-key usage 0x01009B DO_NOT_DISTURB            FALLBACK_USAGE_MAPPING
-key usage 0x0c0065 SCREENSHOT                FALLBACK_USAGE_MAPPING
-key usage 0x0c0067 WINDOW                    FALLBACK_USAGE_MAPPING
-key usage 0x0c006F BRIGHTNESS_UP             FALLBACK_USAGE_MAPPING
-key usage 0x0c0070 BRIGHTNESS_DOWN           FALLBACK_USAGE_MAPPING
-key usage 0x0c0079 KEYBOARD_BACKLIGHT_UP     FALLBACK_USAGE_MAPPING
-key usage 0x0c007A KEYBOARD_BACKLIGHT_DOWN   FALLBACK_USAGE_MAPPING
-key usage 0x0c007C KEYBOARD_BACKLIGHT_TOGGLE FALLBACK_USAGE_MAPPING
-key usage 0x0c00D8 DICTATE                   FALLBACK_USAGE_MAPPING
-key usage 0x0c00D9 EMOJI_PICKER              FALLBACK_USAGE_MAPPING
-key usage 0x0c0173 MEDIA_AUDIO_TRACK         FALLBACK_USAGE_MAPPING
-key usage 0x0c019C PROFILE_SWITCH            FALLBACK_USAGE_MAPPING
-key usage 0x0c019F SETTINGS                  FALLBACK_USAGE_MAPPING
-key usage 0x0c01A2 ALL_APPS                  FALLBACK_USAGE_MAPPING
-key usage 0x0c0201 NEW                       FALLBACK_USAGE_MAPPING
-key usage 0x0c0203 CLOSE                     FALLBACK_USAGE_MAPPING
-key usage 0x0c0208 PRINT                     FALLBACK_USAGE_MAPPING
-key usage 0x0c0227 REFRESH                   FALLBACK_USAGE_MAPPING
-key usage 0x0c0232 FULLSCREEN                FALLBACK_USAGE_MAPPING
-key usage 0x0c029D LANGUAGE_SWITCH           FALLBACK_USAGE_MAPPING
-key usage 0x0c029F RECENT_APPS               FALLBACK_USAGE_MAPPING
-key usage 0x0c02A2 ALL_APPS                  FALLBACK_USAGE_MAPPING
-key usage 0x0d0044 STYLUS_BUTTON_PRIMARY     FALLBACK_USAGE_MAPPING
-key usage 0x0d005a STYLUS_BUTTON_SECONDARY   FALLBACK_USAGE_MAPPING
-
-# Joystick and game controller axes.
-# Axes that are not mapped will be assigned generic axis numbers by the input subsystem.
-axis 0x00 X
-axis 0x01 Y
-axis 0x02 Z
-axis 0x03 RX
-axis 0x04 RY
-axis 0x05 RZ
-axis 0x06 THROTTLE
-axis 0x07 RUDDER
-axis 0x08 WHEEL
-axis 0x09 RTRIGGER
-axis 0x0a LTRIGGER
-axis 0x10 HAT_X
-axis 0x11 HAT_Y
-
-# LEDs
-led 0x00 NUM_LOCK
-led 0x01 CAPS_LOCK
-led 0x02 SCROLL_LOCK
-led 0x03 COMPOSE
-led 0x04 KANA
-led 0x05 SLEEP
-led 0x06 SUSPEND
-led 0x07 MUTE
-led 0x08 MISC
-led 0x09 MAIL
-led 0x0a CHARGING
-
-# SENSORs
-sensor 0x00 ACCELEROMETER X
-sensor 0x01 ACCELEROMETER Y
-sensor 0x02 ACCELEROMETER Z
-sensor 0x03 GYROSCOPE X
-sensor 0x04 GYROSCOPE Y
-sensor 0x05 GYROSCOPE Z
-
-
-key 87   SYM       VIRTUAL
diff --git a/configs/keylayout/keylayout/stmpe_azerty.kl b/configs/keylayout/keylayout/stmpe_azerty.kl
new file mode 100644
index 0000000..bb7b18e
--- /dev/null
+++ b/configs/keylayout/keylayout/stmpe_azerty.kl
@@ -0,0 +1,42 @@
+# Copyright (C) 2015 BlackBerry Limited
+#
+# Blackberry Physical Keyboard
+#
+
+key 5     4
+key 11    0
+key 14    DEL
+key 16    A
+key 17    Z
+key 18    E
+key 19    R
+key 20    T
+key 21    Y
+key 22    U
+key 23    I
+key 24    O
+key 25    P
+key 28    ENTER
+key 30    Q
+key 31    S
+key 32    D
+key 33    F
+key 34    G
+key 35    H
+key 36    J
+key 37    K
+key 38    L
+key 42    SHIFT_LEFT
+key 44    W
+key 45    X
+key 46    C
+key 47    V
+key 48    B
+key 49    N
+key 50    M
+key 54    SHIFT_RIGHT
+key 56    ALT_LEFT
+key 57    SPACE
+key 100   SYM
+key 110   FUNCTION
+key 143   WAKEUP
diff --git a/configs/keylayout/keylayout/stmpe_qwertz.kl b/configs/keylayout/keylayout/stmpe_qwertz.kl
new file mode 100644
index 0000000..efc72e6
--- /dev/null
+++ b/configs/keylayout/keylayout/stmpe_qwertz.kl
@@ -0,0 +1,42 @@
+# Copyright (C) 2015 BlackBerry Limited
+#
+# Blackberry Physical Keyboard
+#
+
+key 5     4
+key 11    0
+key 14    DEL
+key 16    Q
+key 17    W
+key 18    E
+key 19    R
+key 20    T
+key 21    Z
+key 22    U
+key 23    I
+key 24    O
+key 25    P
+key 28    ENTER
+key 30    A
+key 31    S
+key 32    D
+key 33    F
+key 34    G
+key 35    H
+key 36    J
+key 37    K
+key 38    L
+key 42    SHIFT_LEFT
+key 44    Y
+key 45    X
+key 46    C
+key 47    V
+key 48    B
+key 49    N
+key 50    M
+key 54    SHIFT_RIGHT
+key 56    ALT_LEFT
+key 57    SPACE
+key 100   SYM
+key 110   FUNCTION
+key 143   WAKEUP
